inginious/frontend/plugins/register_students/pages/api/add_course_students_csv_file_api.py

```diff
@@ -0,0 +1,149 @@
+
+
+class AddCourseStudentsCsvFile(AdminApi):
+    def API_POST(self):
+        file = get_mandatory_parameter(web.input(), "file")
+        course_id = get_mandatory_parameter(web.input(), "course")
+
+        course = self._get_course_and_check_rights(course_id)
+        if course is None:
+            return 200, {"status": "error", "text": "The course does not exist or the user does not have permissions."}
+
+        text = file.decode("utf-8")
+        parsed_file = self._parse_csv_file(text)
+
+        if not self._file_well_formatted(parsed_file):
+            return 200, {"status": "error", "text": "The file is not formatted as expected, check it is comma separated"
+                                                    " and emails are actual emails."}
+
+        registered_on_course = 0
+        registered_users = 0
+        for user_data in parsed_file:
+            data = self._parse_user_data(user_data)
+
+            result = self._register_student(data)
+            if result:
+                registered_users += 1
+            try:
+                result = self.user_manager.course_register_user(course, data["username"], '', True)
+                if result:
+                    registered_on_course += 1
+            except:
+                pass
+
+        message = "The process finished successfully. Registered students on the course: " + str(registered_on_course) \
+                  + "Registered students in UNCode: " + str(registered_users)
+
+        return 200, {"status": "success", "text": message}
+
+    def _register_student(self, data):
+        """
+        Registers the student in UNCode and sends a verification email to the user. If the user already exists, nothing
+        happens.
+        :param data: Dict containing the user data
+        :return: True if succeeded the register. If user already exists returns False.
+        """
+        success = True
+
+        existing_user = self.database.users.find_one(
+            {"$or": [{"username": data["username"]}, {"email": data["email"]}]})
+        if existing_user is not None:
+            success = False
+        else:
+            passwd_hash = hashlib.sha512(data["passwd"].encode("utf-8")).hexdigest()
+            activate_hash = hashlib.sha512(str(random.getrandbits(256)).encode("utf-8")).hexdigest()
+            self.database.users.insert({"username": data["username"],
+                                        "realname": data["realname"],
+                                        "email": data["email"],
+                                        "password": passwd_hash,
+                                        "activate": activate_hash,
+                                        "bindings": {},
+                                        "language": self.user_manager._session.get("language", "en")})
+            try:
+                web.sendmail(web.config.smtp_sendername, data["email"], "Welcome on UNCode",
```

Comment:
This is ok. But the email could get bigger.

Reply:
What do you mean? could you explain more your comment or the change request.

Reply:
What I mean is that is not considering grow. You could write the mail on a template in case that we need to put more information on this message